### PR TITLE
refactor(cc-tcp-redirection-form): rework state, types and smart for TCP redirection components

### PR DESCRIPTION
**Pull request by hsablonniere**
* apply the new design for component properties with typed states
* use the new defineSmartComponent for the smart definition
* move the skeleton logic to the cc-tcp-redirection component

### CODE REVIEW

File: src/components/cc-tcp-redirection/cc-tcp-redirection.types.d.ts [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+  sourcePort?: number,
 }
 
-export interface Redirection {
-  namespace: string,
-  sourcePort: number,
+export type TcpRedirectionState = TcpRedirectionStateLoading | TcpRedirectionStateLoaded | TcpRedirectionStateWaiting;
+
+interface TcpRedirectionStateLoading {
+  state: 'loading';
```
Comment (pdesoyres-cc) [inline src/components/cc-tcp-redirection/cc-tcp-redirection.types.d.ts:10]: we use double quotes in types.d.ts files

File: src/components/cc-tcp-redirection-form/cc-tcp-redirection-form.types.d.ts [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+
+export type TcpRedirectionFormContextType = "user" | "admin";
+
+export type TcpRedirectionFormState =
+  TcpRedirectionFormStateLoading
+  | TcpRedirectionFormStateLoaded
+  | TcpRedirectionFormStateError;
+
+interface TcpRedirectionFormStateLoading {
+  state: 'loading';
```
Comment (pdesoyres-cc) [inline src/components/cc-tcp-redirection-form/cc-tcp-redirection-form.types.d.ts:11]: we use double quotes in types.d.ts files

File: src/components/cc-tcp-redirection-form/cc-tcp-redirection-form.smart.js [hunk 3]
```diff
@@ -2,83 +2,88 @@ import './cc-tcp-redirection-form.js';
...
+      createTcpRedirection({ apiConfig, ownerId, appId, namespace })
+        .then(({ port }) => {
+          notifySuccess(component, i18n('cc-tcp-redirection-form.create.success', { namespace }));
+          updateRedirection(namespace, (redirection) => {
+            redirection.state = 'loaded';
+            redirection.sourcePort = port;
+          });
+        })
+        .catch(() => {
+          notifyError(component, i18n('cc-tcp-redirection-form.create.error', { namespace }));
```
Comment (pdesoyres-cc) [inline src/components/cc-tcp-redirection-form/cc-tcp-redirection-form.smart.js:43]: add console.error

File: src/components/cc-tcp-redirection-form/cc-tcp-redirection-form.smart.js [hunk 4]
```diff
@@ -2,83 +2,88 @@ import './cc-tcp-redirection-form.js';
...
+      deleteTcpRedirection({ apiConfig, ownerId, appId, sourcePort, namespace })
+        .then(() => {
+          notifySuccess(component, i18n('cc-tcp-redirection-form.delete.success', { namespace }));
+          updateRedirection(namespace, (redirection) => {
+            redirection.state = 'loaded';
+            redirection.sourcePort = null;
+          });
+        })
+        .catch(() => {
+          notifyError(component, i18n('cc-tcp-redirection-form.delete.error', { namespace }));
```
Comment (pdesoyres-cc) [inline src/components/cc-tcp-redirection-form/cc-tcp-redirection-form.smart.js:63]: add console.error

**Review by pdesoyres-cc — CHANGES_REQUESTED** [review]
nice!

**Comment by hsablonniere** [comment]
@pdesoyres-cc Thanks for your feedbacks. It updated the PR :wink:

**Review by florian-sanders-cc — APPROVED** [review]
LGTM :+1: Gg

**Pull request merged**